data/moves.ts

```diff
@@ -14916,10 +14916,20 @@ export const Moves: import('../sim/dex-moves').MoveDataTable = {
 					}
 					// Run through each action in queue to check if the Pursuit user is supposed to Mega Evolve this turn.
 					// If it is, then Mega Evolve before moving.
-					if (source.canMegaEvo || source.canUltraBurst) {
+					if (source.canMegaEvo || source.canUltraBurst || source.canTerastallize) {
 						for (const [actionIndex, action] of this.queue.entries()) {
-							if (action.pokemon === source && action.choice === 'megaEvo') {
-								this.actions.runMegaEvo(source);
```

Comment:
Maybe you could replace this with:
```suggestion
							if (action.pokemon === source && (action.choice === 'megaEvo' || action.choice == 'terastallize')) {
								this.runAction(action);
```

Reply:
> Maybe you could replace this with:

Yeah that would work. `runAction` is basically the same thing as what I wrote originally.